### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by fluiderson**

### CODE REVIEW

**Review by LouiseMedova — APPROVED** [review]

**Review by MedovTimur — APPROVED** [review]

File: io/src/lib.rs [hunk 1]
```diff
@@ -169,3 +175,73 @@ impl From<&NFTState> for IoNFTState {
...
+//         let owners = owner_by_id
+//             .iter()
+//             .map(|(hash, actor_id)| (*actor_id, *hash))
+//             .collect();
+
+//         let token_metadata_by_id = token_metadata_by_id
+//             .iter()
+//             .map(|(id, metadata)| {
+//                 let metadata = metadata.as_ref().unwrap();
+//                 let nft = Nft2 {
```
Comment (shamilsan) [inline io/src/lib.rs]: Dead code?

File: src/contract.rs [hunk 2]
```diff
@@ -158,53 +150,76 @@ unsafe extern "C" fn handle() {
...
-                        exec::program_id(),
-                        payload,
-                        0,
-                        DELAY,
-                    )
+            gstd::debug!(
+                "AZOYAN Update rest_updates_count: {}, token_ids: {:?}",
+                rest_updates_count,
+                token_ids
+            );
```
Comment (shamilsan) [inline src/contract.rs]: Too personally.
```suggestion
            gstd::debug!(
                "Update rest_updates_count: {}, token_ids: {:?}",
                rest_updates_count,
                token_ids
            );
```

File: src/contract.rs [hunk 3]
```diff
@@ -158,53 +150,76 @@ unsafe extern "C" fn handle() {
...
+            nft.update_media(&token_ids);
+            if nft.rest_updates_count == 0 {
+                return;
+            }
+            let action = NFTAction::Update {
+                rest_updates_count: nft.rest_updates_count,
+                token_ids,
+            };
+            let gas_available = exec::gas_available();
+            gstd::debug!("AZOYAN Update. gas_available: {}", gas_available);
```
Comment (shamilsan) [inline src/contract.rs]:
```suggestion
            gstd::debug!("Update. gas_available: {}", gas_available);
```

File: src/contract.rs [hunk 4]
```diff
@@ -249,12 +264,31 @@ impl AutoChangedNft {
...
+                meta.media = media
+            }
+        }
+    }
+    fn reserve_gas(&self) {
+        let reservations = unsafe { &mut RESERVATION };
+        let reservation_id =
+            ReservationId::reserve(RESERVATION_AMOUNT, 600).expect("reservation across executions");
+        reservations.push(reservation_id);
+        // msg::reply(NFTEvent::GasReserved, 0).expect("");
```
Comment (shamilsan) [inline src/contract.rs]:
```suggestion
```

File: src/contract.rs [hunk 5]
```diff
@@ -288,21 +322,76 @@ impl From<&AutoChangedNft> for IoNFT {
...
+        } = value;
+
+        // let transactions = transactions
+        //     .iter()
+        //     .map(|(key, event)| (*key, event.clone()))
+        //     .collect();
+        // let urls = urls
+        //     .iter()
+        //     .map(|(token_id, urls)| (*token_id, urls.clone()))
+        //     .collect();
```
Comment (shamilsan) [inline src/contract.rs]:
```suggestion
```

File: state/src/lib.rs [hunk 6]
```diff
@@ -1,129 +1,126 @@
...
+            base_uri: String::new(),
         }
     }
 
-    pub fn token(state: State, token_id: TokenId) -> Token {
-        token_helper(&token_id, &state)
-    }
+    //     pub fn token(state: State, token_id: TokenId) -> Token {
+    //         token_helper(&token_id, &state)
+    //     }
```
Comment (shamilsan) [inline state/src/lib.rs]:
```suggestion
```

File: state/src/lib.rs [hunk 7]
```diff
@@ -1,129 +1,126 @@
...
+    //         {
+    //             for token_id in token_ids {
+    //                 tokens.push(token_helper(token_id, &state));
+    //             }
+    //         }
+    //         tokens
+    //     }
+    //     pub fn total_supply(state: State) -> u128 {
+    //         state.token.owner_by_id.len() as u128
+    //     }
```
Comment (shamilsan) [inline state/src/lib.rs]: Dead code

**Pull request merged**